src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Delete.cshtml.cs

```diff
@@ -34,12 +34,15 @@ public async Task<IActionResult> OnGetAsync(string id)
 
         public async Task<IActionResult> OnPostAsync(string id)
         {
-            if (id == null)
+            if (User.Identity.IsAuthenticated)
```

Comment:
See if you can use https://docs.microsoft.com/en-us/aspnet/core/security/authorization/razor-pages-authorization?view=aspnetcore-2.2 instead of all the ifs